src/nnet3/nnet-am-decodable-simple.h

```diff
@@ -392,8 +395,247 @@ class DecodableAmNnetSimpleParallel: public DecodableInterface {
+                         int32 output_t_start,
+                         int32 num_output_frames,
+                         Vector<BaseFloat> *ivector);
+
+  // called from constructor
+  void CheckAndFixConfigs();
+
+  // called from AcceptInput()
+  void CheckInput(const Matrix<BaseFloat> *feats,
+                  const Vector<BaseFloat> *ivector = NULL,
+                  const Matrix<BaseFloat> *online_ivectors = NULL);
+
+  // called from AcceptInput() or Compute(). Prepare the batch_info_ which
+  // will be used to compute.
+  void PrepareBatchInfo();
+
+  // called from constructor. According to (tot_left_context,tot_right_context),
+  // which equals to the model left/right context plus the extra left/right
+  // context, we prepare the frequently-used ComputationRequest.
+  // The CompuationRequest will be delete in deconstructor.
+  // Otherwise, we will initialize the "batch_info_" map which is used to
+  // maintain each information entry in (tot_left_context, tot_right_context)
+  // batch. The "batch_info_" map will be delete in deconstructor.
+  void PrepareComputationRequest();
+
+  NnetSimpleComputationOptions opts_;
+  const Nnet &nnet_;
+  int32 nnet_left_context_;
+  int32 nnet_right_context_;
+  int32 output_dim_;
+  // the log priors (or the empty vector if the priors are not set in the model)
+  CuVector<BaseFloat> log_priors_;
+
+  std::unordered_map<std::string, const Matrix<BaseFloat> * > feats_;
+
+  // ivector_ is the iVector if we're using iVectors that are estimated in batch
+  // mode.
+  std::unordered_map<std::string, const Vector<BaseFloat> * > ivectors_;
+
+  // online_ivector_feats_ is the iVectors if we're using online-estimated ones.
+  std::unordered_map<std::string,
+    const Matrix<BaseFloat> * > online_ivector_feats_;
+
+  // online_ivector_period_ helps us interpret online_ivector_feats_; it's the
+  // number of frames the rows of ivector_feats are separated by.
+  int32 online_ivector_period_;
+
+  // an object of CachingOptimizingCompiler. For speed, except for "shorter
+  // than chunk size" batch, we always get the ComputationRequest from
+  // context_to_request_. Then we use CachingOptimizingCompiler to compiler
+  // the CompuationRequest once, when it was first used.
+  CachingOptimizingCompiler compiler_;
+
+  // The current log-posteriors that we got from the last time we
+  // ran the computation.
+  std::unordered_map<std::string, Matrix<BaseFloat>* > log_post_;
+
+  // note: num_subsampled_frames_ will equal feats_.NumRows() in the normal case
+  // when opts_.frame_subsampling_factor == 1.
+  std::unordered_map<std::string, int32> num_subsampled_frames_;
+
+  std::unordered_map<std::string, bool> is_computed_;
+  // store each utterance id in order. We don't use a queue for here as,
+  // in function "compute()" which is a blocking call, we will keep on
+  // computing without taking the results out.
+  std::list<std::string> utt_list_;
+
+  // The meaning of the tuple is
```

Comment:
You should use a struct for this, it will be more readable.  And you need to explain much more clearly what is the meaning of these members-- not jut what they are used for.